lib/evm_database_ops.rb

```diff
@@ -102,6 +104,46 @@ def self.restore(db_opts, connect_opts = {})
     uri
   end
 
+  private_class_method def self.prepare_for_restore(filename)
+    backup_type = validate_backup_file_type(filename)
+
+    if application_connections?
+      message = "Database restore failed. Shut down all evmserverd processes before attempting a database restore"
+      _log.error(message)
+      raise message
+    end
+
+    MiqRegion.replication_type = :none
+    60.times do
+      break if VmdbDatabaseConnection.where("application_name LIKE 'pglogical manager%'").count.zero?
+      _log.info("Waiting for pglogical connections to close...")
+      sleep 5
+    end
+
+    connection_count = backup_type == :basebackup ? VmdbDatabaseConnection.unscoped.count : VmdbDatabaseConnection.count
+    if connection_count > 1
+      message = "Database restore failed. #{connection_count - 1} connections remain to the database."
+      _log.error(message)
+      raise message
+    end
+  end
+
+  private_class_method def self.validate_backup_file_type(filename)
+    if PostgresAdmin.base_backup_file?(filename)
+      :basebackup
+    elsif PostgresAdmin.pg_dump_file?(filename)
+      :pgdump
+    else
+      message = "#{filename} is not in a recognized database backup format"
+      _log.error(message)
+      raise message
+    end
+  end
+
+  private_class_method def self.application_connections?
+    VmdbDatabaseConnection.all.map(&:application_name).any? { |app_name| app_name.start_with?("MIQ") }
```

Comment:
Should `unscoped` be used here too?

Reply:
No, because all it does when used with `.all` is remove the database restriction, but since this one is for application connections we can only query against the configured database.